frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseOverrideForkPointAction.java

```diff
@@ -0,0 +1,144 @@
+import com.virtuslab.gitmachete.backend.api.SyncToParentStatus;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.OverrideForkPointDialog;
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeyGitMacheteRepository;
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeyProject;
+import com.virtuslab.logger.IEnhancedLambdaLogger;
+
+@CustomLog
+public abstract class BaseOverrideForkPointAction extends BaseGitMacheteRepositoryReadyAction
+    implements
+      IExpectsKeyProject,
+      IExpectsKeyGitMacheteRepository,
+      ISyncToParentStatusDependentAction {
+
+  @Override
+  public IEnhancedLambdaLogger log() {
+    return LOG;
+  }
+
+  @Override
+  public @I18nFormat({}) String getActionName() {
+    return getString("action.GitMachete.BaseOverrideForkPointAction.action-name");
+  }
+
+  @Override
+  public @I18nFormat({}) String getDescriptionActionName() {
+    return getString("action.GitMachete.BaseOverrideForkPointAction.description-action-name");
+  }
+
+  @Override
+  public @I18nFormat({GENERAL, GENERAL}) String getEnabledDescriptionFormat() {
+    return getString("action.GitMachete.BaseOverrideForkPointAction.description");
+  }
+
+  @Override
+  public String getCurrentBranchText() {
+    return getString("action.GitMachete.BaseOverrideForkPointAction.text.current-branch");
+  }
+
+  @Override
+  public List<SyncToParentStatus> getEligibleStatuses() {
+    return List.of(SyncToParentStatus.InSyncButForkPointOff);
+  }
+
+  @Override
+  @UIEffect
+  public void onUpdate(AnActionEvent anActionEvent) {
+    super.onUpdate(anActionEvent);
+    syncToParentStatusDependentActionUpdate(anActionEvent);
+  }
+
+  @Override
+  @UIEffect
+  public void actionPerformed(AnActionEvent anActionEvent) {
+    var project = getProject(anActionEvent);
+    var selectedVcsRepository = getSelectedGitRepository(anActionEvent).getOrNull();
+    var branchUnderAction = getNameOfBranchUnderAction(anActionEvent);
+    var branch = branchUnderAction.flatMap(pn -> getGitMacheteBranchByName(anActionEvent, pn)).getOrNull();
+
+    if (selectedVcsRepository == null || branch == null || branch.isRootBranch()) {
+      return;
+    }
+
+    var nonRootBranch = branch.asNonRootBranch();
+    var selectedCommit = new OverrideForkPointDialog(project, nonRootBranch.getParentBranch(), nonRootBranch)
+        .showAndGetSelectedCommit();
+    if (selectedCommit == null) {
+      log().debug(
+          "Selected commit to be the new fork point is null: most likely the action has been canceled from override-fork-point dialog");
+      return;
+    }
+
+    LOG.debug("Enqueueing fork point override");
+    new Task.Backgroundable(project, "Overriding fork point...") {
+      @Override
+      public void run(ProgressIndicator indicator) {
+        overrideForkPoint(anActionEvent, branch, selectedCommit);
+      }
+    }.queue();
+  }
+
+  /**
+   * This method must NOT be called on UI thread.
```

Comment:
Actually, let's start getting that a bit more systematic: pls add sth like `@NotUiThreadSafe` annotation to `qual` module, as for now this will have no effect on compilation... but maybe a tweak to `GuiEffectChecker` won't actually be as difficult  as it seems ;)

Reply:
(coz this: https://github.com/typetools/checker-framework/issues/3253 is unlikely to get addressed anytime soon, alas)